src/components/dashboard/TeamMembersTable/TeamEditInfo.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { useMutation } from '@apollo/client';
+import { useApolloClient } from '@apollo/client';
+import { teamInfo } from 'data/actions/auth';
+import { UPDATE_TEAM_INFO } from '../../../data/gql/team';
+import Toast from 'components/dashboard/Toast/Toast';
+
+import { Box, Text, VStack, Input, Heading, useToast, Button } from '@chakra-ui/react';
+
+const EditTeamInfo = ({ setOnEditPage, teamId, teamName, teamAffiliation }) => {
+  const client = useApolloClient();
+  const [newTeamName, setNewTeamName] = useState(teamName);
+  const [newTeamAffiliation, setNewTeamAffiliation] = useState(teamAffiliation);
+  const [editTeam, { loading: loadingTeamChange, error: teamChangeError }] = useMutation(UPDATE_TEAM_INFO);
+  useEffect(() => {});
+  const dispatch = useDispatch();
+  const toast = useToast();
+
+  const onChangeTeamName = e => {
+    if (e.target.value === '') {
+      setNewTeamName(teamName);
+    } else {
+      setNewTeamName(e.target.value);
+    }
+  };
+
+  const onChangeTeamAffiliation = e => {
+    setNewTeamAffiliation(e.target.value);
+    if (e.target.value === '') {
+      setNewTeamAffiliation(teamAffiliation);
+    } else {
+      setNewTeamAffiliation(e.target.value);
+    }
+  };
+
+  const mutationUpdateTeam = () => {
+    editTeam({ variables: { id: teamId, name: newTeamName, organization: newTeamAffiliation } })
+      .then(() => {
+        dispatch(teamInfo(teamId, client));
+        toast({
+          position: 'top',
+          render: props => <Toast {...props} description="You have updated your team information" isClosable />,
+        });
+      })
+      .catch(e => {
+        console.log(e);
+      });
```

Comment:
we can avoid re-fetching, and instead create a new redux action to update our redux store since you already know what the new team name/team affiliation will be.

Lmk if you need help, take a look at `UPDATE_USER` in `reducers/auth.js` as a template

Lasly, I'd again encourage you to write the `.then` block declaritively using a `useEffect`. You can look thru the codebase, but I'd also first read the react docs on `useEffect` or lmk if u need help